### PR TITLE
Return explicit error regarding build with sandbox/update flags

### LINKED ISSUE #4526: sandbox build with -u option not working
**This is one issue taken from https://github.com/sylabs/singularity/issues/4517**

This issue was reported in the comment from @DrDaveD - and is separate from the first author's problems

### Version of Singularity:

What version of Singularity are you using? Run:

```
3.4.0-1.2
```

### Expected behavior

Can successfully build a sandbox using the ~non-privileged user-namespace~ `-u` option:

*Note - `-u` is for `update` not `userns` when used on `build`*

`singularity build -u --sandbox centos7-sandbox docker://centos:7`

### Actual behavior

Fails with:

```
INFO:    Starting build...
INFO:    Building into existing container: /cloud/login/dwd/scratch/centos7-sandbox
FATAL:   While performing build: failed to retrieve path for /cloud/login/dwd/scratch/centos7-sandbox: lstat /cloud/login/dwd/scratch/centos7-sandbox: no such file or directory
```

### Steps to reproduce this behavior

See above

### PR DESCRIPTION
**Pull request by cclerget**
## Description of the Pull Request (PR):

Due to confusion of `-u/--update` flag, this PR fixes error messages and cover error cases where:
- `--update` is used without `--sandbox` flag
- `--update` is used with `--sandbox` and a non-existent "target" sandbox
- `--update` is used with `--sandbox` and a file as "target" sandbox

In a separate commit there few e2e tests covering those cases.

### This fixes or addresses the following GitHub issues:

 - Fixes #4526

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by jscook2345 — APPROVED** [review]

File: cmd/internal/cli/build.go [hunk 1]
```diff
@@ -278,28 +278,32 @@ func preRun(cmd *cobra.Command, args []string) {
...
 			if err != nil {
-				sylog.Fatalf("Error parsing input: %s", err)
+				return fmt.Errorf("error parsing input: %s", err)
 			}
-			if val := strings.Compare(strings.ToLower(input), "y\n"); val == 0 {
-				force = true
-			} else {
-				sylog.Errorf("Stopping build.")
-				return false
+			if strings.Compare(strings.ToLower(input), "y\n") != 0 {
```
Comment (WestleyK) [inline cmd/internal/cli/build.go]: I think we have a interactive package for this (`internal/pkg/util/interactive`). Should we use that?

**Review by WestleyK — COMMENTED** [review]
Looks good, what about using the `internal/pkg/util/interactive` package for the interactive part?

File: cmd/internal/cli/build.go (hunk 1 above)
Comment (cclerget) [reply cmd/internal/cli/build.go]: Updated

**Review by WestleyK — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**